benchmarks/bench_qselect.hpp

```diff
@@ -0,0 +1,76 @@
+#include "bench-qsort-common.h"
+
+template <typename T>
+static void avx512_qselect(benchmark::State& state) {
+    if (!cpu_has_avx512bw()) {
+        state.SkipWithMessage("Requires AVX512 BW ISA");
+    }
+    if ((sizeof(T) == 2) && (!cpu_has_avx512_vbmi2())) {
+        state.SkipWithMessage("Requires AVX512 VBMI2 ISA");
+    }
+    // Perform setup here
+    size_t ARRSIZE = state.range(0);
+    std::vector<T> arr;
+    std::vector<T> arr_bkp;
+
+    /* Initialize elements */
+    arr = get_uniform_rand_array<T>(ARRSIZE);
+    arr_bkp = arr;
+
+    /* Choose random index to make sorted */
+    int k = get_uniform_rand_array<int64_t>(1, ARRSIZE, 1).front();
```

Comment:
Ditto. Same as partial sort.

Reply:
`avx512_qselect` benchmarks updated to allow `k` to vary.